### PR TITLE
[3.5] Add testing of etcd in local image in release workflow

### PR DESCRIPTION
**Pull request by pchan**
Cherry pick of #15505 on release-3.5.

#15505: scripts: Add testing of etcd in local image in release

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -22,3 +25,6 @@ jobs:
         Expire-Date: 0
         EOF
         DRY_RUN=true ./scripts/release --no-upload --no-docker-push --in-place 3.5.99
+    - name: test-image
+      run: |
+        VERSION=3.6.99 ./scripts/test_images.sh
```
Comment (ahrtr) [inline .github/workflows/release.yaml:30]:
```suggestion
        VERSION=3.5.99 ./scripts/test_images.sh
```

**Comment by ahrtr** [comment]
Please signoff the commit & resolve the workflow failures.. 

Also reminder: usually I do not recommend to automated cherry pick, because 3.5/3.4 have big difference with main branch.

**Comment by pchan** [comment]
Release workflow is now passing. ([link](https://github.com/etcd-io/etcd/actions/runs/4582876927/jobs/8093372537?pr=15608))

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @pchan 

cc @ptabor @serathius

**Review by serathius — APPROVED** [review]

**Pull request merged**